ofo-argo-utils/db_logger.py

```diff
@@ -0,0 +1,127 @@
+    password = os.environ.get("DB_PASSWORD", "")
+    
+    try:
+        conn = psycopg2.connect(
+            host=host,
+            database=database,
+            user=user,
+            password=password
+        )
+        return conn
+    except Exception as e:
+        sys.stderr.write(f"Error connecting to database: {e}\n")
+        sys.exit(1)
+
+def log_datasets_initial(datasets, workflow_id):
+    conn = get_db_connection()
+    try:
+        cursor = conn.cursor()
+        for dataset in datasets:
+            cursor.execute(
+                """
+                INSERT INTO automate_metashape (dataset_name, workflow_id, status) 
+                VALUES (%s, %s, %s) 
+                ON CONFLICT (dataset_name, workflow_id) DO NOTHING
+                """,
+                (dataset, workflow_id, 'queued')
+            )
+        conn.commit()
+    except Exception as e:
+        sys.stderr.write(f"Error logging to database: {e}\n")
+        sys.exit(1)
+    finally:
+        if conn is not None:
+            conn.close()
+
+def log_dataset_start(dataset, workflow_id):
+    conn = get_db_connection()
+    try:
+        cursor = conn.cursor()
+        cursor.execute(
+            """
+            UPDATE automate_metashape 
+            SET status = 'processing', start_time = CURRENT_TIMESTAMP 
+            WHERE dataset_name = %s AND workflow_id = %s
+            """,
+            (dataset, workflow_id)
+        )
+        conn.commit()
+    except Exception as e:
+        sys.stderr.write(f"Error logging start time: {e}\n")
+        sys.exit(1)
+    finally:
+        if conn is not None:
+            conn.close()
+
+def log_dataset_completion(dataset, workflow_id, success):
+    conn = get_db_connection()
+    status = 'completed' if success else 'failed'
+    
+    try:
+        cursor = conn.cursor()
+        cursor.execute(
+            """
+            UPDATE automate_metashape 
+            SET status = %s, finish_time = CURRENT_TIMESTAMP 
+            WHERE dataset_name = %s AND workflow_id = %s
+            """,
+            (status, dataset, workflow_id)
+        )
+        conn.commit()
+    except Exception as e:
+        sys.stderr.write(f"Error logging completion: {e}\n")
+        sys.exit(1)
+    finally:
+        if conn is not None:
+            conn.close()
+
+def main():
+    parser = argparse.ArgumentParser(description='Database logging for Argo workflows')
+    
+    parser.add_argument('action', choices=['log-initial', 'log-start', 'log-completion'],
+                      help='Logging action to perform')
+    
+    parser.add_argument('--workflow-id', required=True,
+                      help='Workflow ID to use in database records')
+    
+    parser.add_argument('--datasets-json',
+                      help='JSON array of dataset names for initial logging')
+    parser.add_argument('--dataset',
+                      help='Dataset name for start/completion logging')
+    parser.add_argument('--success',
```

Comment:
Depending on the usage, this flag could have the `action="store_true"` keyword so it's true if the flag is specified and false otherwise. That may make it trickier to use in a pipeline though.